wfcGen.py

```diff
@@ -0,0 +1,287 @@
+                    cells[flag]["nbg"]["1"].append(cropped[ins][y-1-(res-1)][x])
+                    cells[flag]["nbg"]["2"].append(cropped[ins][y][(x+1+(res-1))%lcx])
+                    cells[flag]["nbg"]["3"].append(cropped[ins][(y+1+(res-1))%lcy][x])
+                else:
+                    cells[flag]["nbg"]["0"].append(cropped[ins][y][x-1])
+                    cells[flag]["nbg"]["1"].append(cropped[ins][y-1][x])
+                    cells[flag]["nbg"]["2"].append(cropped[ins][y][(x+1)%lcx])
+                    cells[flag]["nbg"]["3"].append(cropped[ins][(y+1)%lcy][x])
+            else:
+                if WM:
+                    cells[str(len(cells.keys()))] = {}
+                    cells[str(len(cells.keys())-1)]["nbg"] = {}
+                    cells[str(len(cells.keys())-1)]["nbg"]["0"] = [cropped[ins][y][x-1-(res-1)]]
+                    cells[str(len(cells.keys())-1)]["nbg"]["1"] = [cropped[ins][y-1-(res-1)][x]]
+                    cells[str(len(cells.keys())-1)]["nbg"]["2"] = [cropped[ins][y][(x+1+(res-1))%lcx]]
+                    cells[str(len(cells.keys())-1)]["nbg"]["3"] = [cropped[ins][(y+1+(res-1))%lcy][x]]
+                    cells[str(len(cells.keys())-1)]["img"] = cropped[ins][y][x]
+                else:
+                    cells[str(len(cells.keys()))] = {}
+                    cells[str(len(cells.keys())-1)]["nbg"] = {}
+                    cells[str(len(cells.keys())-1)]["nbg"]["0"] = [cropped[ins][y][x-1]]
+                    cells[str(len(cells.keys())-1)]["nbg"]["1"] = [cropped[ins][y-1][x]]
+                    cells[str(len(cells.keys())-1)]["nbg"]["2"] = [cropped[ins][y][(x+1)%lcx]]
+                    cells[str(len(cells.keys())-1)]["nbg"]["3"] = [cropped[ins][(y+1)%lcy][x]]
+                    cells[str(len(cells.keys())-1)]["img"] = cropped[ins][y][x]
+
+    for y in range(lcy):
+        for x in range(lcx):
+            for key in cells.keys():
+                if cells[key]["img"] == cropped[ins][y][x]:
+                    Gkey = key
+            for key in cells.keys():
+                for i in range(len(cells[key]["nbg"]["0"])):
+                    if cells[key]["nbg"]["0"][i] == cropped[ins][y][x]:
+                        cells[key]["nbg"]["0"][i] = Gkey
+                for i in range(len(cells[key]["nbg"]["1"])):
+                    if cells[key]["nbg"]["1"][i] == cropped[ins][y][x]:
+                        cells[key]["nbg"]["1"][i] = Gkey
+                for i in range(len(cells[key]["nbg"]["2"])):
+                    if cells[key]["nbg"]["2"][i] == cropped[ins][y][x]:
+                        cells[key]["nbg"]["2"][i] = Gkey
+                for i in range(len(cells[key]["nbg"]["3"])):
+                    if cells[key]["nbg"]["3"][i] == cropped[ins][y][x]:
+                        cells[key]["nbg"]["3"][i] = Gkey
+
+    print(cells)
+
+def intersection(mass1, mass2):
+    return list(set(mass1) & set(mass2))
+
+class Cell:

```

Comment:
It is hard to read this code because you first have something in the procedural style (for loops on the most global level), and then a class definition. It's better to define classes as functions first. And, for that matter, not have any global logic at all (keep everything in functions and classes). But we can leave the latter for a future pull request